.travis.yml

```diff
@@ -5,53 +5,87 @@ stages:
   - name: doc
     if: branch = master AND type != pull_request
 
-before_install:
-  - echo -e "machine github.com\n  login $CI_USER\n  password $CI_USER_PASSWORD" >> ~/.netrc
+env:
+  global:
+    - PYENV_VERSION=3.6
+    - CHANS_DEV="-c pyviz/label/dev -c ioam -c bokeh -c conda-forge -c erdc -c defaults"
+    - CHANS_REL="-c pyviz -c erdc -c defaults"
+
+cache:
+  directories:
+    - $HOME/miniconda
+
+before_cache:
+  - rm -rf $HOME/miniconda/pkgs
+  - rm -rf $HOME/miniconda/conda-bld/*
+
+# quick hack to determine what tag is (improvements welcomed)
+#     release: ^v(\d+|\.)*[^a-z]\d*$
+# dev release: ^v(\d+|\.)*[a-z]\d*$
+
+stages:
+  - test
+  - name: conda_dev_package
+    if: tag =~ ^v(\d+|\.)*[a-z]\d*$
+  - name: conda_package
+    if: tag =~ ^v(\d+|\.)*[^a-z]\d*$
 
 jobs:
   include:
+
+    ########## TEST DEVELOPER INSTALL ##########
+
+    # quick tests in typical dev env (most tests are run on packages)
+
     - &default
       stage: test
-      os: linux
+      before_install:
+        # install doit/pyct and use to install miniconda...
+        - pip install pyct && doit miniconda_install && pip uninstall -y doit pyct
+        - export PATH="$HOME/miniconda/bin:$PATH" && hash -r
+        - conda config --set always_yes True
+        # ...and now install doit/pyct into miniconda
+        - conda install -c pyviz/label/dev pyct && doit ecosystem_setup
       install:
-        #######################
-        ## install miniconda ##
-        - easy_install --user doit==0.29.0 ioamdoit
-        - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-            ~/Library/Python/2.7/bin/doit install_miniconda;
-          else
-            ~/.local/bin/doit install_miniconda;
-          fi
-        - rm -f .doit.db
-        - export PATH="$HOME/miniconda/bin:$PATH"
-        #######################
-        - pip install ioamdoit
-        - conda env create
-        - source activate earthsim
-        - doit develop_install
-        - doit capture_conda_env
+        - doit env_create $CHANS_DEV --python=$PYENV_VERSION
+        - source activate test-environment
+        - doit develop_install $CHANS_DEV -o recommended
+        - doit env_capture
       script:
-        - doit download_sample_data
-        - travis_wait 60 doit all_tests
+        - earthsim download_data --path=examples
+        - travis_wait 60 doit test_all
+
+      # TODO: restore macos if travis has recovered
 
-# restore when travis/macos recovers...
 #    - <<: *default
-#      os: osx
-#      before_install:
-#        # brew-installed geos interferes with cartopy?
-#        - brew uninstall --ignore-dependencies geos gdal postgis
-#      if: branch = master AND type != pull_request
+#      stage: doc
+#      script:
+#        - doit install_doc_dependencies
+#        - doit download_sample_data
+#        - travis_wait 60 doit docs
+#      deploy:
+#        edge:
+#          branch: v1.8.47
```

Comment:
Where does the string "v1.8.47" come from?

Reply:
That should be a comment in the file, not here in the PR, right? It stands out, and we'll want to be able to remove it with confidence later when it's fixed.